src/Disks/DiskWebServer.h

```diff
@@ -0,0 +1,174 @@
+
+    String getFileName(const String & path) const;
+
+    DiskType::Type getType() const override { return DiskType::Type::WebServer; }
+
+    std::unique_ptr<ReadBufferFromFileBase> readFile(const String & path,
+                                                     size_t buf_size,
+                                                     size_t estimated_size,
+                                                     size_t aio_threshold,
+                                                     size_t mmap_threshold,
+                                                     MMappedFileCache * mmap_cache) const override;
+    /// Disk info
+
+    const String & getName() const final override { return name; }
+
+    const String & getPath() const final override { return metadata_path; }
+
+    UInt64 getTotalSpace() const final override { return std::numeric_limits<UInt64>::max(); }
+
+    UInt64 getAvailableSpace() const final override { return std::numeric_limits<UInt64>::max(); }
+
+    UInt64 getUnreservedSpace() const final override { return std::numeric_limits<UInt64>::max(); }
+
+    /// Read-only part
+
+    bool exists(const String & path) const override;
+
+    bool isFile(const String & path) const override;
+
+    size_t getFileSize(const String & path) const override;
+
+    void listFiles(const String & /* path */, std::vector<String> & /* file_names */) override { }
+
+    void setReadOnly(const String & /* path */) override {}
+
+    bool isDirectory(const String & path) const override;
+
+    DiskDirectoryIteratorPtr iterateDirectory(const String & /* path */) override;
+
+    Poco::Timestamp getLastModified(const String &) override { return Poco::Timestamp{}; }
+
+    /// Write and modification part
+
+    std::unique_ptr<WriteBufferFromFileBase> writeFile(const String &, size_t, WriteMode) override;
+
+    void moveFile(const String &, const String &) override {}
+
+    void replaceFile(const String &, const String &) override {}
+
+    void removeFile(const String &) override
+    {
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Disk {} is read-only", getName());
+    }
+
+    void removeFileIfExists(const String &) override
+    {
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Disk {} is read-only", getName());
+    }
+
+    ReservationPtr reserve(UInt64 /*bytes*/) override
+    {
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Disk {} is read-only", getName());
+    }
+
+    void removeRecursive(const String &) override
+    {
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Disk {} is read-only", getName());
+    }
+
+    void removeSharedFile(const String &, bool) override {}
+
+    void removeSharedRecursive(const String &, bool) override {}
+
+    void clearDirectory(const String &) override {}
+
+    void moveDirectory(const String &, const String &) override {}
+
+    void removeDirectory(const String &) override {}
+
+    void setLastModified(const String &, const Poco::Timestamp &) override {}
```

Comment:
All such methods should throw `NOT_IMPLEMENTED`. I think it's bad idea to silently do nothing, because caller expects that operation is performed successfully if no exceptions thrown. Also it's unclear why some methods throw and others does not.